File: src/main/java/com/pyzy/server808/decoder/JTT808BasedFrameEncoder.java
package com.pyzy.server808.decoder;

import com.pyzy.server808.message.JTTMessage;
import com.pyzy.server808.message.Message;
import io.netty.buffer.ByteBuf;
import io.netty.channel.ChannelHandlerContext;
import io.netty.handler.codec.MessageToByteEncoder;

public class JTT808BasedFrameEncoder extends MessageToByteEncoder<JTTMessage> {
    @Override
    protected void encode(ChannelHandlerContext ctx, JTTMessage msg, ByteBuf out) throws Exception {
        ByteBuf buf = msg.encoder();
        out.writeBytes("7e".getBytes());
        Byte bytes = Message.Companion.xorCheckCode(buf,buf.readableBytes());
        out.writeBytes(buf);
        out.writeByte(bytes);
        out.writeBytes("7e".getBytes());
    }
}

File: src/main/java/com/pyzy/server808/decoder/JTT808BasedFrameEncoder.kt
package com.pyzy.server808.decoder

import com.pyzy.server808.ext.printHexString
import com.pyzy.server808.message.JTTMessage
import com.pyzy.server808.message.Message
import io.netty.buffer.ByteBuf
import io.netty.channel.ChannelHandlerContext
import io.netty.handler.codec.MessageToByteEncoder

class JTT808BasedFrameEncoder : MessageToByteEncoder<Message<JTTMessage>>() {
    @Throws(Exception::class)
    override fun encode(ctx: ChannelHandlerContext, msg: Message<JTTMessage>, out: ByteBuf) {

        val bytes = msg.encoder()
        println("终端注册响应: ");bytes.printHexString()
        out.writeBytes(bytes)
    }
}
